### PR TITLE
feat(contracts): update colosseum for Cancun upgrade

### PR DESCRIPTION
**Pull request by 0xHansLee**
This includes upgrade of Colosseum contract according to Cancun upgrade.
Challenge data is also updated. 

FYI, I regenerated bindings with the latest version of forge.
The version is `forge 0.2.0 (f625d0f 2024-04-02T00:16:42.810315000Z)`

### CODE REVIEW

**Comment by 0xHansLee** [comment]
@coderabbitai review this PR please.

File: kroma-bindings/bindings/colosseum_more.go [hunk 1]
```diff
@@ -13,7 +13,7 @@ const ColosseumStorageLayoutJSON = "{\"storage\":[{\"astId\":1000,\"contract\":\
...
...10190565b60008151602080840160005b838110156156775781518752958201959082019060010161565b565b509495945050505050565b8c81528b60208201528a604082015289606082015288608082015260007fffffffffffffffff000000000000000000000000000000000000000000000000808a60c01b1660a0840152808960c01b1660a88401528760b0840152808760c01b1660d0840152507fffff0000000000000000000000000000000000000000000000000000000000008560f01b1660d883015261572961572360da84018661564f565b8461564f565b9e9d5050505050505050505050505050565b6000835161574d818460208801614841565b835190830190615761818360208801614841565b01949350505050565b600181815b808511156157c357817fffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff048211156157a9576157a9614dec565b808516156157b657918102915b93841c939080029061576f565b509250929050565b6000826157da575060016115ca565b816157e7575060006115ca565b81600181146157fd576002811461580757615823565b60019150506115ca565b60ff84111561581857615818614dec565b50506001821b6115ca565b5060208310610133831016604e8410600b8410161715615846575081810a6115ca565b615850838361576a565b807fffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff0482111561588257615882614dec565b029392505050565b6000612ae583836157cb56fea164736f6c634300080f000a"
```
[the remark this answers is not in the document]
Comment (0xHansLee) [reply kroma-bindings/bindings/colosseum_more.go:16]: This is auto-generated file, so I'll skip this review

**Review by sm-stack — COMMENTED** [review]
How about adding an extra upgrade tx for Colosseum contract at `op-node/rollup/derive/ecotone_upgrade_transactions.go`, like other upgraded contracts in Ecotone upgrade?

**Comment by 0xHansLee** [comment]
> How about adding an extra upgrade tx for Colosseum contract at op-node/rollup/derive/ecotone_upgrade_transactions.go, like other upgraded contracts in Ecotone upgrade?

Colosseum is L1 contract, so we can do that.

**Review by Pangssu — APPROVED** [review]
I don't know why all bindigns have been changed, but it doesn't matter for the behavior, so I approve.

**Comment by kangsorang** [comment]
It would be nice to have consideration for before and after ecotone upgrades, as it currently stands it seems like it should be at least a week after the ecotone upgrade.

File: packages/contracts/contracts/libraries/Types.sol [hunk 2]
```diff
@@ -66,6 +66,9 @@ library Types {
         bytes32 stateRoot;
         bytes32 withdrawalsRoot;
         bytes32[] txHashes;
+        uint64 blobGasUsed;
```
[the remark this answers is not in the document]
Comment (0xHansLee) [reply packages/contracts/contracts/libraries/Types.sol:72]: Added in [71794d9](https://github.com/kroma-network/kroma/pull/291/commits/71794d967df0c134e0308f1092ad0e8688f12807)

**Review by kangsorang — APPROVED** [review]

**Pull request merged**